packages/lodestar/src/chain/chain.ts

```diff
@@ -121,6 +123,7 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
     await this.blockProcessor.start();
     this._currentForkDigest =  computeForkDigest(this.config, state.fork.currentVersion, state.genesisValidatorsRoot);
     this.on("forkDigestChanged", this.handleForkDigestChanged);
+    await this.restoreHeadState(state);
```

Comment:
why not call this inside waitForState?

Reply:
we have to do this after forkchoice.start() and probably blockProcessor.start()